azkaban-exec-server/src/main/java/azkaban/execapp/FlowRunner.java

```diff
@@ -1268,7 +1268,9 @@ public void kill(final String user) {
 
   public void kill() {
     synchronized (this.mainSyncObj) {
-      if (isKilled()) {
+      if (isKilled() || this.flowFinished) {
```

Comment:
Will this.flowFinished verify with every finished state stated in Status.isStatusFinished method? If so, no need to verify isKilled.

Reply:
No, `flowFinished` is an in-memory boolean attribute which is set to true when a flow execution is finalized.

Reply:
Thank you for explaining that.